### PR TITLE
Don't scope course searches by provider for course change support page

### PR DESCRIPTION
**Pull request by elliotcm**
This allows support staff to move a candidate from one provider to another, e.g. when organisations merge etc.

## Context

We're doing this as a (probably) short term measure to allow the merging of several providers.

### Before
Screenshot 2022-09-28 at 12 40 16

### After
Screenshot 2022-09-28 at 12 31 28

## Link to Trello card

https://trello.com/c/pZbvttyX/686-relax-provider-only-restriction-when-changing-courses
https://trello.com/c/GykXgxhQ/1715-transfer-applicants-to-different-organisation

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by elliotcm** [comment]
[Relax provider only restriction when changing courses](https://trello.com/c/pZbvttyX/686-relax-provider-only-restriction-when-changing-courses)

**Comment by carlosmartinez** [comment]
looks good. let's create a review app and get declan/others to confirm they're happy.

**Comment by elliotcm** [comment]
After a chat with the support team we've come up with a couple of quick ways to make this clearer and less error prone, so I'll add that in now.

**Review by carlosmartinez — APPROVED** [review]
great. looks like a better solution.

File: app/forms/support_interface/application_forms/pick_course_form.rb [hunk 1]
```diff
@@ -96,6 +88,18 @@ def courses_for_provider(provider)
...
       end
 
+      def courses_for_other_providers(provider)
+        return Course.none if provider.blank?
+
+        courses - courses_for_provider(provider)
+      end
+
+      def course_radio_options_for(courses)
+        courses.map { |course|
```
Comment (carlosmartinez) [inline app/forms/support_interface/application_forms/pick_course_form.rb:98]: rubocop says multiline blocks use `do` and `end` 🤷🏽‍♂️

File: app/forms/support_interface/application_forms/pick_course_form.rb [hunk 2]
```diff
@@ -86,7 +78,7 @@ def courses
       end
 
       def courses_for_provider(provider)
-        return [] if provider.blank?
+        return Course.none if provider.blank?
```
Comment (JonF1) [inline app/forms/support_interface/application_forms/pick_course_form.rb:81]: what's the benefit of using `Course.none` over empty array?

**Review by JonF1 — APPROVED** [review]

File: app/forms/support_interface/application_forms/pick_course_form.rb (hunk 2 above)
Comment (JonF1) [reply app/forms/support_interface/application_forms/pick_course_form.rb:81]: Thanks, that's helpful to know 👍

**Pull request merged**